lib/shared/bucketing-assembly-script/assembly/types/dvcUser.ts

```diff
@@ -33,6 +65,44 @@ export class DVCUser extends JSON.Obj implements DVCUserInterface {
         super()
     }
 
+    static fromPBUser(userPB: DVCUser_PB): DVCUser {
+        const nullableEmail = userPB.email
+        const nullableName = userPB.name
+        const nullableLanguage = userPB.language
+        const nullableCountry = userPB.country
+        const nullableAppBuild = userPB.appBuild
+        const nullableAppVersion = userPB.appVersion
+        const nullableDeviceModel = userPB.deviceModel
+        const nullableCustomData = userPB.customData
+        const nullablePrivateCustomData = userPB.privateCustomData
+
+        return new DVCUser(
+            userPB.userId,
+            (nullableEmail && !nullableEmail.isNull) ? nullableEmail.value : null,
+            (nullableName && !nullableName.isNull) ? nullableName.value : null,
+            (nullableLanguage && !nullableLanguage.isNull) ? nullableLanguage.value : null,
+            (nullableCountry && !nullableCountry.isNull) ? nullableCountry.value : null,
+            (nullableAppBuild && !nullableAppBuild.isNull) ? nullableAppBuild.value : NaN,
+            (nullableAppVersion && !nullableAppVersion.isNull) ? nullableAppVersion.value : null,
+            (nullableDeviceModel && !nullableDeviceModel.isNull) ? nullableDeviceModel.value : null,
+            getJSONObjFromPBCustomData(nullableCustomData),
+            getJSONObjFromPBCustomData(nullablePrivateCustomData),
+        )
+    }
+
+    toProtoBuf(): Uint8Array {
+        return encodeDVCUser_PB(new DVCUser_PB(
```

Comment:
can we not directly use this DVCUser_PB object rather than translating back and forth to our existing dvcUser class? or is that too hard to integrate with the existing code?

I have a feeling we'd gain even more by not having to copy all this memory every time

Reply:
Potentially yes, but I think it would require some extensive changes since we had to do a lot of munging to fit these objects into how protobuf can represent them

Reply:
yea protobuf can't really do nulls easily, so we need these translation layers. We can look into optimizations later, but I don't think they will help much.

Reply:
also this method is only used for testing.